src/views/emails/template-parts/body/order/order-gateway-data.php

```diff
@@ -28,6 +29,11 @@
 	return;
 }
 
+// No need to show gateway ID if it's the same as the order ID.
+if ( is_numeric( $order->gateway_order_id ) && intval( $order->ID ) === intval( $order->gateway_order_id ) ) {
+	return;
+}
```

Comment:
This logic is a bit more complex than necessary, there could be other gateways where the `gateway_order_Id` could be same as `Order_id`, let's just skip this for free orders depending on the `total_value` as well.

Reply:
In my opinion, if the gateway order is the same as the order id (whether it's free or not), then this information is redundant and shouldn't be shown. What is the reason we're not showing the Gateway ID for free orders? An ID exists, right? If the reasoning is that we don't need to show it due to the order ID being shown elsewhere in the email template, then this logic should suffice.